ts/src/bingx.ts

```diff
@@ -0,0 +1,2317 @@
+        //                  "minNotional": 5,
+        //                  "maxNotional": 20000,
+        //                  "status": 1,
+        //                  "tickSize": 0.000001,
+        //                  "stepSize": 1
+        //                },
+        //              ]
+        //            }
+        //    }
+        //
+        const result = [];
+        const data = this.safeValue (response, 'data');
+        const markets = this.safeValue (data, 'symbols', []);
+        for (let i = 0; i < markets.length; i++) {
+            result.push (this.parseMarket (markets[i]));
+        }
+        return result;
+    }
+
+    async fetchSwapMarkets (params) {
+        const response = await this.swapV2PublicGetQuoteContracts (params);
+        //
+        //    {
+        //        "code": 0,
+        //        "msg": "",
+        //        "data": [
+        //          {
+        //            "contractId": "100",
+        //            "symbol": "BTC-USDT",
+        //            "size": "0.0001",
+        //            "quantityPrecision": 4,
+        //            "pricePrecision": 1,
+        //            "feeRate": 0.0005,
+        //            "tradeMinLimit": 1,
+        //            "maxLongLeverage": 150,
+        //            "maxShortLeverage": 150,
+        //            "currency": "USDT",
+        //            "asset": "BTC",
+        //            "status": 1
+        //          },
+        //        ]
+        //    }
+        //
+        const result = [];
+        const markets = this.safeValue (response, 'data');
+        for (let i = 0; i < markets.length; i++) {
+            result.push (this.parseMarket (markets[i]));
+        }
+        return result;
+    }
+
+    parseMarket (market) {
+        const id = this.safeString (market, 'symbol');
+        const symbolParts = id.split ('-');
+        const baseId = symbolParts[0];
+        const quoteId = symbolParts[1];
+        const base = this.safeCurrencyCode (baseId);
+        const quote = this.safeCurrencyCode (quoteId);
+        const currency = this.safeString (market, 'currency');
+        const settle = this.safeCurrencyCode (currency);
+        let pricePrecision = this.safeInteger (market, 'pricePrecision');
+        if (pricePrecision === undefined) {
+            pricePrecision = this.precisionFromString (this.safeString (market, 'tickSize'));
+        }
+        let quantityPrecision = this.safeInteger (market, 'quantityPrecision');
+        if (quantityPrecision === undefined) {
+            quantityPrecision = this.precisionFromString (this.safeString (market, 'stepSize'));
+        }
+        const type = (settle !== undefined) ? 'swap' : 'spot';
+        const spot = type === 'spot';
+        const swap = type === 'swap';
+        let symbol = base + '/' + quote;
+        if (settle !== undefined) {
+            symbol += ':' + settle;
+        }
+        const contractSize = this.safeNumber (market, 'size');
```

Comment:
market does not seem to have that `size` field 
![image](https://github.com/ccxt/ccxt/assets/43336371/1e151176-676d-4a5d-ac37-3a9d1ee5cfed)![image](https://github.com/ccxt/ccxt/assets/43336371/7f1b4a26-0fd8-41a5-b6ad-1043933cbc64)
![Uploading image.png…]()

Reply:
@carlosmiei The swap markets have the `size` field

Reply:
sorry I though I was parsing a swap market